src/coreclr/jit/optimizebools.cpp

```diff
@@ -1339,10 +1339,13 @@ void OptBoolsDsc::optOptimizeBoolsUpdateTrees()
     }
 
     // Update loop table
-    m_comp->fgUpdateLoopsAfterCompacting(m_b1, m_b2);
-    if (optReturnBlock)
+    if (m_comp->optLoopTableValid)
     {
-        m_comp->fgUpdateLoopsAfterCompacting(m_b1, m_b3);
+        m_comp->fgUpdateLoopsAfterCompacting(m_b1, m_b2);
+        if (optReturnBlock)
+        {
+            m_comp->fgUpdateLoopsAfterCompacting(m_b1, m_b3);
+        }
```

Comment:
Some of these checks are always false, like in this case... I'll remove them all as part of the big code deletion in the future.